java-frontend/src/main/java/org/sonar/java/resolve/Resolve.java

```diff
@@ -575,30 +578,35 @@ private boolean isArgumentsAcceptable(List<JavaType> argTypes, List<JavaType> fo
       ArrayJavaType lastFormal = (ArrayJavaType) formals.get(formalsSize - 1);
       JavaType argType = argTypes.get(argsSize - i);
       // check type of element of array or if we invoke with an array that it is a compatible array type
-      if (!isAcceptableType(argType, lastFormal.elementType, autoboxing) && (nbArgToCheck != 1 || !isAcceptableType(argType, lastFormal, autoboxing))) {
+      if (!isAcceptableType(env, argType, lastFormal.elementType, autoboxing) && (nbArgToCheck != 1 || !isAcceptableType(env, argType, lastFormal, autoboxing))) {
         return false;
       }
     }
     for (int i = 0; i < argsSize - nbArgToCheck; i++) {
-      if (!isAcceptableType(argTypes.get(i), formals.get(i), autoboxing)) {
+      JavaType arg = argTypes.get(i);
+      JavaType formal = formals.get(i);
+      if (!isAcceptableType(env, arg, formal, autoboxing)) {
         return false;
       }
     }
     return true;
   }
 
-  private boolean isAcceptableType(JavaType arg, JavaType formal, boolean autoboxing) {
+  private boolean isAcceptableType(Env env, JavaType arg, JavaType formal, boolean autoboxing) {
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Reduce the number of returns of this method 6, down to the maximum allowed 5. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://dory.sonarsource.com/sonarqube/coding_rules#rule_key=squid%3AS1142)